### ISSUE TITLE
File Explorer Panel does not update

### ISSUE THREAD

**Issue by jtsom** [Microsoft/vscode#69149]
Issue Type: <b>Bug</b>

The file explorer panel, including the Source Control panel, does not update consistantly when files change. Switching branches via the terminal sometimes leaves the old branch's file in the window. Also, the branch in the lower left corner does not reflect the new branch. If there are files that have changed, the source control panel does not show file updates. I have to constantly click the Refresh button to see changes.

VS Code version: Code - Insiders 1.32.0-insider (2e4d4a6bb16a65303b460795fa4ac734f9aea43e, 2019-02-21T06:14:37.115Z)
OS version: Darwin x64 18.2.0

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz (12 x 2600)|
|GPU Status|2d_canvas: enabled<br>checker_imaging: disabled_off<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: enabled<br>rasterization: enabled<br>surface_synchronization: enabled_on<br>video_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 2, 2|
|Memory (System)|32.00GB (0.60GB free)|
|Process Argv||
|Screen Reader|no|
|VM|25%|

</details><details><summary>Extensions (16)</summary>

Extension|Author (truncated)|Version
---|---|---
ng-template|Ang|0.1.11
path-intellisense|chr|1.4.2
vscode-eslint|dba|1.8.0
githistory|don|0.4.6
auto-rename-tag|for|0.0.15
csharpextensions|jch|1.3.0
Angular2|joh|7.0.1
azure-account|ms-|0.8.0
csharp|ms-|1.18.0-beta7
vscode-typescript-tslint-plugin|ms-|1.0.0
debugger-for-chrome|msj|4.11.2
vetur|oct|0.16.2
material-icon-theme|PKi|3.6.3
quicktype|qui|12.0.46
open-in-browser|tec|2.0.0
vscodeintellicode|Vis|1.1.3

(3 theme extensions excluded)

</details>

**Comment by jtsom** [comment]
Not a duplicate of any of those.

**Comment by mikerockett** [comment]
This has been happening on my install as well – ever since the most recent stable update.

**Comment by Lakret** [comment]
The same for me, since the latest update (Version 1.31.1 (1.31.1), macOS Mojave 10.14.1) renamed files are not renamed in the sidebar, new files added with external programs do not appear in the sidebar, and  newly created files don't appear there either. Manual refresh helps, but that's def. a bug.

**Comment by jtsom** [comment]
An additional note/observation - it doesn't matter if the "Experimental File Watcher" is enabled or disabled.

**Comment by birkir** [comment]
Looks like it is fixed with latest Insiders (1.32)

**Comment by jtsom** [comment]
(this may be another issue entirely)
Switching branches from a Terminal panel (git checkout master .. git checkout another-branch), while it looks like the file tree does update, the branch indicator in the lower left does not update to the current branch. Off to file another report...

**Comment by jtsom** [comment]
Not sure if it's the same issue... but updating a file, and saving, still does not show any updates in the source control view. Have to manually click the Refresh button to see the changes.

**Comment by joaomoreno** [comment]
@jtsom Did you open the root folder of the repository with VS Code, or a subfolder of the repo?

**Comment by jtsom** [comment]
Yes, it is a folder within the repo. (the .git directory is up one level). I do notice that if I wait, the source control panel will update, but it takes several minutes.

**Comment by joaomoreno** [comment]
Duplicate of #41085

**Comment by jtsom** [comment]
Great, duplicate of a year plus old issue. I guess I just have to get used to clicking the Refresh button.

**Comment by mikerockett** [comment]
@joaomoreno I'm not sure how this is the case, considering that this is a behaviour change after the most recent update… In my case, I use workspaces with two or more git repos in them. Still having to click refresh.

**Comment by joaomoreno** [comment]
@MikeChou If you are indeed opening a sub folder of a git repo, it's certainly not new. This is a long standing issue, blocked by https://github.com/Microsoft/vscode/issues/3025. Please upvote that issue to raise its awareness.

**Comment by JakeHL** [comment]
Not sure if this is useful, but I've been experiencing this issue only since the latest update. It happens regardless of where I am in the git repo, root or otherwise.

Let me know if there are any logs I can provide.

**Comment by rmehner** [comment]
Happens for me as well, my `.git` folder is in the current folder. Have to reload all the time. Enabled experimental file watcher, but no change.

VSCode 1.31.1 on macOS 10.14.3.

Let me know if there's anything more you need :)

**Comment by nguyendviet** [comment]
Happens to me now. Since this one is closed, I opened one here: https://github.com/Microsoft/vscode/issues/69851
